### ISSUE TITLE
Merging files from multiple locations

### ISSUE THREAD

**Issue by quiescer** [gtimelog/gtimelog#189]
Thanks for Gtimelog - I find it strikes a good balance between simplicity and convenience. One feature that I've been thinking about is having the ability to synch time logs from 2 (or more?) computers at different locations.

With Covid I've been working from home a lot, and it's sometimes difficult to keep track of time when it's spread over multiple computers.

Perhaps somebody has a script to intelligently merge two timelogs? I assume if there are no overlaps, it could probably be quite a straightforward process.

I also saw mention of using a remote file via HTTP and would be very interested to know more about that option!

**Comment by quiescer** [comment]
I guess something like:

**Comment by mgedmin** [comment]
I've heard of people putting their `~/.gtimelog` folder in Dropbox or a similar synchronization product.

**Comment by StephaneMangin** [comment]
I will propose a Jira integration soon, for this i have to modify a little bit the Task class and it's integration in main process to allow multiple files to process... incoming soon (i use it since few weeks and i wait to see if it's really stable)

**Comment by mgedmin** [comment]
What sort of Jira integration?  I'm currently using a script my coworker wrote to parse timelog.txt and update Jira ticket worklog entries: https://github.com/ProgrammersOfVilnius/gtimelog2jira

I'm thinking eventually it would be nice for gtimelog to support plugins so this sort of thing could be done from the UI.

**Comment by StephaneMangin** [comment]
> What sort of Jira integration? I'm currently using a script my coworker wrote to parse timelog.txt and update Jira ticket worklog entries: https://github.com/ProgrammersOfVilnius/gtimelog2jira
> 
> I'm thinking eventually it would be nice for gtimelog to support plugins so this sort of thing could be done from the UI.

You're right, a little bit of modularity would be much appreciable (i think especially of extracting email management, remote tasks file and of course Jira tasks retrieval or all other API based integration). The separation of UI and core code is not as forward pushed i think too. Maybe it could be great to freeze a little bit the codebase and do this job together with others. What you think ?

So far, be able to configure in the UI (only ?) a task retrieval taking care of all the possible configuration of such integration would be a pain in the ass imo. Be able to juste extend a class as a module to allow new integration would be much simpler and much sustainable in time i presume (using a king of interface for that or just a declaration based module using a builder).

In my job too, we parse the tasklog to process in jira (it works great and may be enough in major use i think). Sadly, the code is private and part of a lot of automation scripts so i can't share it.

By the way, i takes lot of fun coding in gtimelog (many challenges due to the current architecture ;) ), maybe my work would be usefull, or not ^^

**Comment by mgedmin** [comment]
Yes, I expect the code will need to be refactored to allow for plugins.  main.py is getting too large anyways, I want to split it into several modules.

**Comment by thevikas** [comment]
Yes. plugin support would be great. Also if possible, try if it can be language independent. Like installing plugins from config file, etc. Cause I made this [cmd line utility](https://github.com/thevikas/gtimelog-php-cli) in php. I have to make invoices  every month around these days. So I will soon automate this in php.

**Comment by jpsnyder** [comment]
Plugin support for different integrations would be nice. I made a very rudimentary script to upload my gtimelog entries to Toggl, and would love to clean it up to integrate it into the application itself.

Many people have to record their timesheet into whatever software suite their company decided to purchase without much say. Having an easy way to add integration into many different timesheet services, supported by the community would be huge.
